fe/fe-core/src/main/java/com/starrocks/load/ExportJob.java

```diff
@@ -290,8 +290,11 @@ private void plan(ExportStmt stmt) throws UserException {
 
                     Long assignedBytes = bytesPerBe.get(backendId);
                     if (assignedBytes == null || assignedBytes < maxBytesPerBe) {
-                        taskTabletLocations.add(scanRangeLocations);
-                        bytesPerBe.put(backendId, assignedBytes != null ? assignedBytes + dataSize : dataSize);
+                        // if dataSize == 0, skip this scanRange
+                        if (dataSize > 0) {
```

Comment:
dataSize depend on BE report, may be inconsistent with actual size.

Reply:
In some bad cases this may be a bug, i maintain the behaviour as the original export, it still export the empty file.
In most cases, a tablet will not be empty, the cost of exporting empty file can be ignored